module/VuFind/src/VuFind/RecordDriver/MarcAdvancedTrait.php

```diff
@@ -951,4 +951,57 @@ public function getCleanNBN()
         }
         return false;
     }
+
+    /**
+     * Get the full title of the record in alternative scripts.
+     *
+     * @return array
+     */
+    public function getTitlesAltScript(): array
+    {
+        return $this->getMarcReader()
+            ->getLinkedFieldsSubfields('880', '245', ['a', 'b']);
```

Comment:
It may be impractical due to the way the logic is currently spread across multiple traits (and perhaps this is a sign that the design could be improved), but I wonder if it would be useful to create properties for holding the MARC definitions, so if we redefine which fields/subfields are used for any of these things, we only have to change it in one place in order to affect both the "regular" and 880-based methods. (I'm thinking, in particular, about things like Unimarc support, which might be easier if we could simply override some properties instead of having to rewrite all the methods). Obviously, I'm happy to merge this as-is when you are ready and think about that larger issue later... but this brought it to mind.

Reply:
It would make sense, but I'm afraid it won't be easy. I suppose there could be a definition file MARC 21 and architecture that would allow you to implement e.g. UNIMARC support similarly. But there are cases that depend on indicator values, fields that are alternatives etc. that make such definition difficult to build. For simple cases such as "result list title subfields" and "record view title subfields", that would work, but anything more complex, I'm not sure.

I'd like to get feedback from @bbusenius before marking this ready for merging.

Reply:
Agreed, a simple map couldn't solve all problems... but even if it got us 80% of the way there, it would make it less intimidating to create a SolrUnimarc subclass, since we would only have to override methods with significant functional differences. In any case, a conversation for another day! :-)

I'll await feedback from @bbusenius before proceeding further.

Reply:
@EreMaijala this is beautiful! I feel like you built a lot of versatility into this with things like the `concat` argument for `getLinkedFieldsSubfields` and the ability to simply get all subfields or specify them in the `getLinkedFields`. I think those subtleties might prove to be useful. I can see how we could override the methods in `MarcAdvancedTrait` if we wanted to change any of the defaults. I can't speak much to what @demiankatz is saying about UNIMARC support but from what I'm seeing here, I feel like this would work for us.